xbmc/dbwrappers/Database.cpp

```diff
@@ -688,7 +700,7 @@ void CDatabase::RollbackTransaction()
 
 bool CDatabase::InTransaction()
 {
-  if (NULL != m_pDB.get()) return false;
```

Comment:
This seems to be a bug? We return `false` if `m_pDB` is non-NULL, otherwise defererence the null pointer.

Reply:
It looks really like a bug :laughing:.

`m_pDB->in_transaction();` should be called after that, in this case never, or if called then with `nullptr` where it must bring a crash :stuck_out_tongue_closed_eyes:

Have looked in the history here on git, is since Kodi is on github. A very very old bug.